### DISCUSSION TITLE
Airflow times out on Long Running Kubernetes Jobs with logging gaps

### DISCUSSION THREAD

**Discussion by alittlesliceoftom** [apache/airflow#26482] [General]
This is listed too in other locations:

https://issues.apache.org/jira/browse/AIRFLOW-3534
https://issues.apache.org/jira/browse/AIRFLOW-4526

Only known fix is to disable log stream: 
`When disabling the `get_logs` parameter everything works because the log stream is skipped.`

Suggested fix - up the retry count on the below line, or allow this to be configured. 
https://github.com/apache/airflow/blob/b23fc137812f5eabf7834e07e032915e2a504c17/airflow/kubernetes/pod_launcher.py#L198

**Comment by alittlesliceoftom** [comment]
Happy to raise this as a PR, I.e. changing 
` @tenacity.retry(stop=tenacity.stop_after_attempt(3), wait=tenacity.wait_exponential(), reraise=True) `
to
` @tenacity.retry(stop=tenacity.stop_after_attempt(5), wait=tenacity.wait_exponential(), reraise=True) `
Which I think would solve the issue for most people and make this much more usable, but I'm not used to this code base!

**Comment by alittlesliceoftom** [comment]
(note my version exp issue was 1.1.14

**Comment by alittlesliceoftom** [comment]
Note, I tried this fix and it **didn't work**. The best thread I have found is this one: 

https://github.com/apache/airflow/pull/7428#issuecomment-704994507

**Comment by eladkal** [comment]
Does this happens with latest Airflow version?

**Comment by potiuk** [comment]
Converted it to discussion. We have revamped K8S integration since quite heavily and the log streaming part was specifically rewritten using more stable and newer k8s library so this issue is not relevant any more most likely (we can always reopen if it is in the latest version - so please upgrade and double-check @alittlesliceoftom

**Comment by Ans-hrfc** [comment]
Hi @potiuk , still happening with Airflow `2.5.1` running in composer

**Reply by potiuk** [reply]
I am not sure why 2.5.1 when 2.5.3 (with only bugfixes) is there for many months.  I suggest you describe all details, logs, circumstances etc. - first to composer team and see wha they tell. They provide paid support and have specific configuration in their K8S so maybe they will be able to find the reason.

**Reply by lorenh516** [reply]
Also happening with Airflow `2.9.3` running in composer for various circumstances/ DAGs.

**Reply by potiuk** [reply]
Well. If you can provide more details and logs and open an issue about it that will allow someone to reproduce it or at least guess what could be reason, please do that. Otherwise "various circumstances/DAGs" are hardly making it possible for anyone who would try to diagnose it.
